src/include/migraphx/serialize.hpp

```diff
@@ -102,8 +103,15 @@ auto to_value_impl(rank<10>, const T& x) -> decltype(x.to_value())
     return x.to_value();
 }
 
+template <class T, MIGRAPHX_REQUIRES(std::is_same<T, shape::type_t>{})>
+value to_value_impl(rank<11>, const T& x)
```

Comment:
Why is this overload added? The enum overload [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/serialize.hpp#L88) should handle this.

Reply:
Without the overload, we need a constructor for the value with an enum as the input...